alibi_detect/utils/_types.py

```diff
@@ -37,3 +34,5 @@
 # type aliases, for use with mypy (must be FwdRef's if involving opt. deps.)
 OptimizerTF: TypeAlias = Union['tf.keras.optimizers.Optimizer', 'tf.keras.optimizers.legacy.Optimizer',
                                Type['tf.keras.optimizers.Optimizer'], Type['tf.keras.optimizers.legacy.Optimizer']]
+
+TorchDeviceType: TypeAlias = Optional[Union[Literal['cuda', 'gpu', 'cpu'], 'torch.device']]
```

Comment:
re the forward reference `'torch.device'` in here, I can't think of a good fix at the moment, but just noting that this  introduces lots of additional sphinx warnings, and is not rendered "perfectly" in the docs (we've gone from 6 to 29 warnings, which makes me sad).

I suspect the forward ref would be resolved during docs compilation if we installed `alibi-detect[all]` on read-the-docs (https://github.com/SeldonIO/alibi-detect/pull/499) which is now allowed, but it seems wasteful...

Reply:
:face_exhaling: arg... I'll open an issue. Maybe this PR might need to be reigned in!